### ISSUE TITLE
Library size is large

### ISSUE THREAD

**Issue by andrewmclagan** [WP-API/node-wpapi#450]
The size of this library is large

https://bundlephobia.com/result?p=wpapi@1.2.1

Simply for a api-sdk client. Anyway we could reduce it?

**Comment by kadamwhite** [comment]
Definitely; several, actually. There are other issues talking about precomputing the available endpoint methods from the JSON endpoint response, which would remove a decent number of raw bytes for the bundle. There’s other opportunities for improvement as well. 

That said, because the interface we’ve chosen to expose requires a decent amount of information about the host system and the API in general, this is never going to be as lightweight as a pure HTTP library.

**Comment by andrewmclagan** [comment]
Yes; My thoughts also when looking into this. It's more an architecture than dependancy graph issue. Seems having allot of the "sugar" that allows devs to have a fluent interface to the JSON API adds some considerable size.

As an alternate offering, considering this size and such, I have created a pull request (very large, a basic rewrite) on another popular WP JSON API client. If you're interested in taking a look. Basically its goal is to come in at under 5Kb and have still offer some nice feature. Having an abstrated transport layer helps allot: 

View PR here: Maybe we can share some goals, code and learnings. Having two offerings with different approaches can be good for the ecosystem. (Very early days this PR)

https://github.com/ylletjs/yllet/pull/29

**Comment by andrewmclagan** [comment]
We are currently around `2.bKb`

https://bundlephobia.com/result?p=@yllet/client@2.0.0-alpha.2

for alpha version
